vermouth/tests/test_name_moltype.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2018 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Test for the NameMolType processor.
+"""
+
+from collections import defaultdict
+import itertools
+from glob import glob
+import os.path
+import subprocess
+import pytest
+from hypothesis import given, assume
+from hypothesis import strategies as st
+from vermouth import System
+from vermouth.processors.name_moltype import NameMolType
+from .molecule_strategies import random_molecule
+from .datafiles import PDB_HB
+
+@st.composite
+def molecules_and_moltypes(draw, max_moltypes=4, min_size=0, max_size=None,
```

Comment:
Move this to molecule_strategies.py
Or maybe upgrade molecule_strategies.py to just strategies.py (or make it a submodule) and collect all strategies there.

Reply:
This strategy is extremely *ad-hoc* for the test that comes just after: it produces the test case and the expected result for that specific test. I can move the strategy to `molecule_strategies.py`, but I do not like having the test split in two files. It is not an issue for the other strategies in `molecule_strategies.py` because they are more generic.

Reply:
My concern is that all our tests are ad-hoc and causing code and effort duplication

Reply:
If we need this strategy elsewhere, we will have to consider that the new code being testing may itself be redundant.